src/coreComponents/physicsSolvers/fluidFlow/StencilOutput.cpp

```diff
@@ -0,0 +1,233 @@
+{
+  ConnectionData conn;
+  for( localIndex i = 0; i < 2; ++i )
+  {
+    conn.transmissibility[i] = kernelData.transmissibility[i];
+    conn.globalId[i] = localToGlobalMap[kernelData.regionId[i]][kernelData.subRegionId[i]][kernelData.elementId[i]];
+  }
+  return conn;
+}
+
+
+void StencilOutput::outputTimeStepData( MeshLevel const & mesh,
+                                        string_view stencilName,
+                                        real64 const outputTime,
+                                        arrayView1d< KernelConnectionData > const & kernelData )
+{
+  string const fileName = getOutputFileName( stencilName, "csv" );
+  bool const newFile = outputFiles.count( fileName ) == 0;
+  std::ofstream outputFile( fileName, newFile ? std::ios_base::out : std::ios_base::app );
+
+  if( newFile )
+  {
+    writeCsvHeader( outputFile );
+    outputFiles.insert( fileName );
+    GEOS_LOG_RANK_IF( getLogLevel() > 0, GEOS_FMT( "{}: initialized file \"{}\".",
+                                                   getDataContext(), fileName ) );
+  }
+
+  { // output timestep data
+    std::set< ConnectionData > fileData;
+
+    LocalToGlobalMap localToGlobalMap = mesh.getElemManager().constructArrayViewAccessor< globalIndex, 1 >(
+      ObjectManagerBase::viewKeyStruct::localToGlobalMapString() );
+
+    std::transform( kernelData.begin(), kernelData.end(), std::inserter( fileData, fileData.end() ),
+                    [&]( auto const & kernelConn )
+    {
+      return ToConnectionData( kernelConn, localToGlobalMap );
+    } );
+
+    writeCsvData( outputFile, fileData, outputTime );
+    GEOS_LOG_RANK_IF( getLogLevel() > 0, GEOS_FMT( "{}: {} connections writen in file \"{}\".",
+                                                   getDataContext(), kernelData.size(), fileName ) );
+  }
+
+  outputFile.close();
+}
+
+
+string StencilOutput::getOutputFileName( string_view stencilName, string_view extension ) const
+{
+  return GEOS_FMT( "{}/{}.{}",
+                   m_outputDir, stencilName, extension );
+}
+
+void StencilOutput::writeCsvHeader( std::ofstream & outputFile ) const
+{
+  outputFile << GEOS_FMT( "{},A global id,B global id,{},{}",
+                          units::getDescription( units::Unit::Time ),
+                          GEOS_FMT( "A {}", units::getDescription( units::Unit::Transmissibility ) ),
+                          GEOS_FMT( "B {}", units::getDescription( units::Unit::Transmissibility ) ) );
+  outputFile << std::endl;
+}
+
+template< typename CONN_DATA_CONT >
+void StencilOutput::writeCsvData( std::ofstream & outputFile, CONN_DATA_CONT const & data, real64 const time ) const
+{
+  for( StencilOutput::ConnectionData const & conn : data )
+  {
+    outputFile << time << ","
+               << conn.globalId[0] << ","
+               << conn.globalId[1] << ","
+               << conn.transmissibility[0] << ","
+               << conn.transmissibility[1] << std::endl;
```

Comment:
I would avoid the flush that is implied by `std::endl` and use `\n` - it might speed things up a fraction.